x/txfees/client/cli/query.go

```diff
@@ -0,0 +1,142 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/osmosis-labs/osmosis/x/txfees/types"
+	"github.com/spf13/cobra"
+)
+
+// GetQueryCmd returns the cli query commands for this module
+func GetQueryCmd(queryRoute string) *cobra.Command {
+	// Group queries under a subcommand
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("Querying commands for the %s module", types.ModuleName),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	cmd.AddCommand(
+		GetCmdFeeTokens(),
+		GetCmdDenomPoolID(),
+		GetCmdBaseDenom(),
+	)
+
+	return cmd
+}
+
+// GetCmdGaugeIds takes the pool id and returns the matching gauge ids and durations
+func GetCmdFeeTokens() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "fee-tokens",
+		Short: "Query the list of non-basedenom fee tokens and their associated pool ids",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Query the list of non-basedenom fee tokens and their associated pool ids
+
+Example:
+$ %s query txfees fee-tokens
+`,
+				version.AppName,
+			),
+		),
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			res, err := queryClient.FeeTokens(cmd.Context(), &types.QueryFeeTokensRequest{})
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdGaugeIds takes the pool id and returns the matching gauge ids and durations
+func GetCmdDenomPoolID() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "denom-pool-id",
+		Short: "Query the pool id associated with a specific whitelisted fee token",
```

Comment:
If ATOM is whitelisted for ATOM/OSMO and ATOM/AKT pair, what does this return?

Reply:
It will check that only one pool can be whitelisted per denom.  It also should make sure that selected pool does contain the native fee token (OSMO)